.jestSetup.js

```diff
@@ -5,7 +5,30 @@ if (
   typeof globalThis.TextEncoder === "undefined" ||
   typeof globalThis.TextDecoder === "undefined"
 ) {
-  const utils = require("util");
-  globalThis.TextEncoder = utils.TextEncoder;
-  globalThis.TextDecoder = utils.TextDecoder;
+  const utils = require("util")
+  globalThis.TextEncoder = utils.TextEncoder
+  globalThis.TextDecoder = utils.TextDecoder
 }
+
+jest.mock(`gatsby-worker`, () => {
+  const gatsbyWorker = jest.requireActual(`gatsby-worker`)
+
+  const { WorkerPool: OriginalWorkerPool } = gatsbyWorker
+
+  class WorkerPoolThatCanUseTS extends OriginalWorkerPool {
+    constructor(workerPath, options) {
+      options.env = {
+        ...(options.env ?? {}),
+        NODE_OPTIONS: `--require ${require.resolve(
+          `./packages/gatsby/src/utils/worker/__tests__/test-helpers/ts-register.js`
+        )}`,
+      }
+      super(workerPath, options)
+    }
+  }
+
+  return {
+    ...gatsbyWorker,
+    WorkerPool: WorkerPoolThatCanUseTS,
+  }
+})
```

Comment:
This is just a trick for unit tests to be able to use source TS files when using `gatsby-worker` / child process setup

In actual usage TS source files become CJS and those will be used (all e2e tests would be using that for example)